gtsam/navigation/Gal3ImuEKF.h

```diff
@@ -0,0 +1,107 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file  Gal3ImuEKF.h
+ * @brief Extended Kalman Filter for IMU-driven Gal3
+ *
+ * @date  September 2025
+ * @authors Scott Baker
+ */
+
+#pragma once
+
+#include <gtsam/navigation/LeftLinearEKF.h>  // Include the base class
+#include <gtsam/geometry/Gal3.h>
+#include <gtsam/navigation/PreintegrationParams.h>
+
+namespace gtsam {
+
+/// Specialized EKF for IMU-driven on Gal3
+class GTSAM_EXPORT Gal3ImuEKF : public LeftLinearEKF<Gal3> {
+ public:
+  using Base = LeftLinearEKF<Gal3>;
+  using TangentVector = typename Base::TangentVector;  // Vector9
+  using Jacobian = typename Base::Jacobian;            // 9x9
+  using Covariance = typename Base::Covariance;        // 9x9
+
+  /**
+   * Construct with initial state/covariance and preintegration params (for
+   * gravity and IMU covariances)
+   * @param X0 Initial Gal3.
+   * @param P0 Initial covariance in tangent space at X0.
+   * @param params Preintegration parameters providing gravity and options.
+   */
+  Gal3ImuEKF(const Gal3& X0, const Covariance& P0,
+                 const std::shared_ptr<PreintegrationParams>& params);
+
+  /// Calculate W (gravity-only left composition, world-frame increments)
```

Comment:
Seems like this copy/paste could be resolved with a small "mix-in" class, maybe `BaseImuEKF`